core/server/common/src/main/java/alluxio/master/journal/JournalUpgrader.java

```diff
@@ -0,0 +1,268 @@
+import java.util.List;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Tool to upgrade journal from v0 to v1. Format the v1 journal before running this tool.
+ *
+ * It is strongly recommended to backup the v0 journal before running this tool to avoid losing
+ * any data in case of failures.
+ *
+ * <pre>
+ * java -cp assembly/target/alluxio-assemblies-0.9.0-SNAPSHOT-jar-with-dependencies.jar \
+ *   alluxio.master.journal.JournalUpgrader -journalDirectoryV0 YourJournalDirectoryV0
+ * </pre>
+ */
+@NotThreadSafe
+public final class JournalUpgrader {
+  private static final Logger LOG = LoggerFactory.getLogger(JournalUpgrader.class);
+
+  private static final int EXIT_FAILED = -1;
+  private static final int EXIT_SUCCEEDED = 0;
+  private static final Options OPTIONS =
+      new Options().addOption("help", false, "Show help for this test.")
+          .addOption("journalDirectoryV0", true,
+              "Where the v0 journal is persisted. It is assumed to be the same as the v1 journal "
+                  + "directory if not set.");
+
+  private static boolean sHelp;
+  private static String sJournalDirectoryV0;
+
+  /**
+   * A class that provides a way to upgrade the journal from v0 to v1.
+   */
+  private static final class Upgrader {
+    private final String mMaster;
+    private final alluxio.master.journalv0.MutableJournal mJournalV0;
+    private final Journal mJournalV1;
+
+    private final UnderFileSystem mUfs;
+
+    private final URI mCheckpointV0;
+    private final URI mCompletedLogsV0;
+
+    private final URI mCheckpointsV1;
+    private final URI mLogsV1;
+
+    private Upgrader(String master) {
+      mMaster = master;
+      mJournalV0 = (new alluxio.master.journalv0.MutableJournal.Factory(
+          getJournalLocation(sJournalDirectoryV0))).create(master);
+      mJournalV1 = (new Journal.Factory(
+          getJournalLocation(Configuration.get(PropertyKey.MASTER_JOURNAL_FOLDER)))).create(master);
+
+      mUfs = UnderFileSystem.Factory.get(sJournalDirectoryV0);
+
+      mCheckpointV0 = URIUtils.appendPathOrDie(mJournalV0.getLocation(), "checkpoint.data");
+      mCompletedLogsV0 = URIUtils.appendPathOrDie(mJournalV0.getLocation(), "completed");
+
+      mCheckpointsV1 = URIUtils.appendPathOrDie(mJournalV1.getLocation(), "checkpoints");
+      mLogsV1 = URIUtils.appendPathOrDie(mJournalV1.getLocation(), "logs");
+    }
+
+    /**
+     * Upgrades journal from v0 to v1.
+     *
+     * @throws IOException if any I/O errors occur
+     */
+    void upgrade() throws IOException {
+      if (!mUfs.exists(mCheckpointV0.toString())) {
+        LOG.info("No checkpoint is found for {}. No upgrade is required.", mMaster);
+        return;
+      }
+      prepare();
+
+      LOG.info("Starting to upgrade {} journal.", mMaster);
+      boolean checkpointUpgraded = false;
+      int logNumber = 1;
+      URI completedLog;
+      while (mUfs.exists((completedLog = getCompletedLogV0(logNumber)).toString())) {
```

Comment:
we do not need to because it is handled by 'journalWriterV0.completeLogs();' in line 152